Разметка всех страниц

YaTyulen opened the pull request:
---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2327229/tasks/5)

:boom: https://htmlacademy-htmlcss.github.io/2327229-device-36/2/

YaTyulen committed:
Добавляет хедер

YaTyulen committed:
Добавляет секцию со слайдами в мейне

YaTyulen committed:
Добавляет секцию с превью категорий магазина

YaTyulen committed:
Добавляет секцию с доп. услугами

YaTyulen committed:
Добавление заголовка в секцию с категориями товаров

YaTyulen committed:
Добавляет секцию с кнопкой доставки под заказ

YaTyulen committed:
Добавляет секцию "О нас"

YaTyulen committed:
Добавляет класс кнопке из секции доставка под заказ

YaTyulen committed:
Исправляет класс у кнопок в секции дополнительных услуг

Чтобы у всех подобных кнопок на сейте был одинаковый класс

YaTyulen committed:
Добавляет секцию с контактами

YaTyulen committed:
Исправляет кнопку на ссылку в секции О нас

YaTyulen committed:
Добавляет секцию с подпиской на рассылку

YaTyulen committed:
Добавляет футер

YaTyulen committed:
Добавление ссылок на соцсети в футере

YaTyulen committed:
Оборачивает логотип html.academy в футере в ссылку

YaTyulen committed:
Добавляет ссылки для блока с категориями товаров

YaTyulen committed:
Создает страницу каталога

YaTyulen committed:
Добавление хедера на страницу каталога

YaTyulen committed:
Исправление ссылок на кнопки в хэдере и добавление блока div в хэдере

YaTyulen committed:
Добавление футера в страницу каталога

YaTyulen committed:
Изменение заголовка страницы каталога

YaTyulen committed:
Добавляет main на страницу каталога

YaTyulen committed:
Исправление имени группы радиокнопок

YaTyulen committed:
Изменение ссылки "Назад" на неактивную

Поскольку это первая страница

YaTyulen committed:
Добавление плейсхолдера в поле в хэдере

YaTyulen committed:
Добавление name в поле для ввода в хэдере

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+      <nav class="navigation">
 
-    <p>Репозиторий создан для обучения на профессиональном онлайн‑курсе «<a href="https://htmlacademy.ru/intensive/htmlcss">HTML и CSS. Профессиональная вёрстка сайтов</a>».</p>
+        <section class="logo">
+          <a href="#">
+            <img class="image-logo" src="" alt="Логотип магазина «Девайс»">
+          </a>
+        </section>
```
Отдельно логотип в секцию оборачивать не нужно. Это перебор).

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+          </a>
+        </section>
+
+        <section class="user-tools">
+          <div class="search-and-login">
+            <form action="https://echo.htmlacademy.ru/" class="search-form" method="post">
+              <input class="search-input" type="text" name="site-search" placeholder="Поиск по сайту">
+              <button class="search-button" type="submit">Найти</button>
+            </form>
+            <div class="header-user">
+              <img class="user-icon" src="" alt="">
+              <a class="button-login" href="#">Войти</a>
+            </div>
+          </div>
+          <ul class="navigation-user">
+            <li class="navigation-user-item">
+              <button class="navigation-user-button">Сравнить</button>
+            </li>
+            <li class="navigation-user-item">
+              <button class="navigation-user-button">Корзина</button>
+            </li>
+          </ul>
+        </section>
```
В 18 стр. `<div class="search-and-login">` - давай эту обертку пока уберем. В данный момент выглядит как избыточная. На этапе стилизации при необходимости можно будет подкорекритровать. search-and-login - не оч хорошее название. Желаительно не привязывать в нейминге классов к содержимому секции.

Этот кусочек надо разметить также списком, несмотря на то, что там всего один пункт. Пукнты могут добавиться в будущем.
```
<div class="header-user">
      <img class="user-icon" src="" alt="">
      <a class="button-login" href="#">Войти</a>
 </div>
```

```
<section class="user-menu">
     <form action="https://echo.htmlacademy.ru/" class="search-form" method="post">
          <label></label> - в паре с инпутом всегда идет лейбл. Надо добавить
          <input class="search-input" type="text" name="site-search" placeholder="Поиск по сайту">
          <button class="search-button" type="submit">Найти</button>
      </form>
      <ul class="user-navigation">
          <li class="user-navigation-item">
              <a class="button-login" href="#">Войти</a>
          </li>
      <ul>
      <ul class="site-navigation">
        <li class="user-navigation-item">
          <button class="navigation-user-button">Сравнить</button>
        </li>
        <li class="user-navigation-item">
          <button class="user-navigation-button">Корзина</button>
        </li>
      </ul>
</section>
```

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+        <section class="site-navigation">
+          <button class="button-drop-menu">
+            <span>Каталог товаров</span>
+            <img src="" alt="Кнопка выпадающего меню">
```
`<img src="" alt="Кнопка выпадающего меню">` Иконку потом лучше будет добавить через css.

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+
+    <main class="page-main">
+      <h1 class="visually-hidden">Магазин "Девайс"</h1>
+      <section class="sliders">
```
`<section class="sliders">` Лучше в ед числе. Слайдер один

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+      <section class="sliders">
+        <h2 class="visually-hidden">Презентация товаров</h2>
+
+        <ol class="slider-pagination">
+          <li class="slider-pagination-item">
```
` <ol class="slider-pagination"> ` pagination - обычно используется в нейминге касаемо пагинации. Т.е переключения страниц в приложении. Лучше назвать `slider-items`. Соответсвенно внутри списка лишка - это один слайд `slider-tem`

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+
+        <ol class="slider-pagination">
+          <li class="slider-pagination-item">
+            <div class="left-slider">
+              <img src="" alt="Квадракоптер" class="slider-image">
+            <button class="slider-button slider-prev" type="button">
+              <span class="visually-hidden">Предыдущий слайд</span>
+            </button>
+            <button class="slider-button slider-next" type="button">
+              <span class="visually-hidden">Следующий слайд</span>
+            </button>
+            </div>
+            <div class="right-slider">
+              <strong>Порхает как бабочка,
+                жалит как пчела!</strong>
+                <p>Этот обычный, на первый взгляд, квадрокоптер оснащен мощным лазером, замаскированным под стандартную камеру.</p>
+                <button class="button-slider-item" type="button">Подробнее</button>
+                <table class="characteristic-product">
+                  <tr>
+                    <td>800 м</td>
+                    <td>50 м</td>
+                  </tr>
+                  <tr>
+                    <th>Дальность полета</th>
+                    <th>Радиус поражения</th>
+                  </tr>
+                </table>
+            </div>
+          </li>
```
Несмотря на визуал, на странице нету левого или правого слайда. Слайд - один). Есть картинка, контролы и описание. 
Контролы (стрелки лево - право) - не должны дублироваться. 

```
   <button class="slider-button slider-prev" type="button">
      <span class="visually-hidden">Предыдущий слайд</span>
    </button>
    <button class="slider-button slider-next" type="button">
      <span class="visually-hidden">Следующий слайд</span>
    </button>
```
Контролы общие на весь слайдер. Также еще есть три точки, это тоже контролы и они тоже не должны дублироваться. В разметке кстати их надо доразметить 

![image](https://user-images.githubusercontent.com/77338210/217364387-72f51a7d-d899-442e-b2fb-987b603fc1a3.png)

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+                правильно!</strong>
+                <p>Мотивирующие фитнес-браслеты помогут найти в себе силы не пропускать занятия и соблюдать диету.</p>
+                <button class="button-slider-item" type="button">Подробнее</button>
+                <table class="characteristic-product">
+                  <tr>
+                    <td>48 часов</td>
+                  </tr>
+                  <tr>
+                    <th>Без подзарядки</th>
+                  </tr>
+                </table>
```
Оч не советую юзать тут таблицы. Замучаешься на стилизации. Можно разметить дивами и спанами

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+
+      </section>
+
+      <section class="Additional-terms">
```
`Additional-terms` все с маленькой буквы `additional-terms`

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+
+      </section>
+
+      <section class="Additional-terms">
+        <h2 class="visually-hidden">Дополнительные услуги</h2>
+        <div class="left-box">
+          <ul class="services-list">
+            <li class="services-item">
+              <button class="button button-delivery">Доставка</button>
+            </li>
+            <li class="services-item">
+              <button class="button button-guarantee">Гарантия</button>
+            </li>
+            <li class="services-item">
+              <button class="button button-credit">Кредит</button>
+            </li>
+          </ul>
+        </div>
+        <div class="right-box">
+          <h3>Доставка</h3>
+          <p>Мы с удовольствием доставим ваш товар прямо
+            к вашему подъезду совершенно бесплатно! Ведь
+            мы неплохо заработаем, поднимая его на ваш этаж.</p>
+            <img src="" alt="Курьер" class="courier-icon">
+        </div>
+
+      </section>
```
В секции additional-terms также есть проблема с неймингом. Привязка к расположению `left-box` `right-box`. Надо дать абстрактные имена

`<img src="" alt="Курьер" class="courier-icon">` скорее декоративная иконка. Лучше задать через css

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+      <section class="delivery-on-order">
+        <h2 class="visually-hidden">Доставка под заказ</h2>
+        <button class="button-delivery-on-order">
+          <img src="" alt="Иконка доставки">
```
`<img src="" alt="Иконка доставки">` - тоже надо в css

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+        </button>
+      </section>
+
+      <section class="information-company">
+        <h2>О нас</h2>
+        <p>Огромный выбор гаджетов не оставит равнодушным гика, который есть в каждом из нас.</p>
+        <p>Мы можем доставить ваш товар в самые отдалённые уголки России транспортными компаниями:</p>
+        <ul class="transport-companies-list">
+          <li class="transport-companies-item">Меловые линии</li>
+          <li class="transport-companies-item">СДИК</li>
+          <li class="transport-companies-item">ТоксБерри</li>
+        </ul>
+        <a href="#" class="button">Подробнее о нас</a>
+      </section>
+
+      <section class="contacts">
+        <h2>Контакты</h2>
+        <p>Вы можете забрать товар сами, заехав в наш офис.
+          Заодно сможете проверить работоспособность покупки.</p>
+          <h3 class="contacts-subtitle">Интернет-магазин «Девайс»</h3>
+          <address class="contacts-address">
+            <p>Санкт-Петербург, набережная реки Карповки, 5, литера П.</p>
+            <a class="contacts-phone" href="tel:+78128121212">+7 (812) 812-12-12</a>
+          </address>
+          <h3 class="contacts-subtitle">Время работы</h3>
+          <ul class="contacts-schedule">
+            <li class="contacts-schedule-item">пн—пт: с 10:00 до 22:00</li>
+            <li class="contacts-schedule-item">сб—вс: с 10:00 до 19:00</li>
+          </ul>
+          <a href="#" class="button">Напишите нам</a>
+      </section>
```
`<section class="information-company">` и `<section class="contacts">` оберни в див или доп section

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+            <button class="button newsletter-button" type="submit">Подписаться</button>
+          </form>
+          <div class="addition">
+            <img src="" alt="">
```
`<img src="" alt="">` - иконочку через стили(css).

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+    </main>
+
+    <footer class="page-footer">
+      <section class="footer-address-section">
+        <h2 class="visually-hidden">Адрес</h2>
+        <img class="footer-address-logo" src="" alt="Логотип интернет-магазина Девайс">
+        <address class="footer-address">
+          <p>Санкт-Петербург, набережная реки Карповки, 5, литера П.</p>
+        </address>
+      </section>
```
```
<section class="footer-address-section">
<section class="footer-links">
<section class="footer-contacts">
```

Оборачивать все в секции в футере как хедере немного избыточно) футер как бы сам по себе раздел. Лого с маленьким текстом на целую секцию не тянет. Можно оставить, можно использовать обычный див для обертки

yury-dergachev commented on index.html:
```diff
@@ -2,11 +2,326 @@
+    <footer class="page-footer">
+      <section class="footer-address-section">
+        <h2 class="visually-hidden">Адрес</h2>
+        <img class="footer-address-logo" src="" alt="Логотип интернет-магазина Девайс">
```
`<img class="footer-address-logo" src="" alt="Логотип интернет-магазина Девайс">` - думаю тут тоже нужно обернуть в ссылку

yury-dergachev commented on catalog.html:
```diff
@@ -0,0 +1,257 @@
+            <form action="https://echo.htmlacademy.ru/" class="search-form" method="post">
+              <input class="search-input" type="text" name="site-search" placeholder="Поиск по сайту">
+              <button class="search-button" type="submit">Найти</button>
+            </form>
+            <div class="header-user">
+              <img class="user-icon" src="" alt="">
+              <span class="user-name">Коля Петухов</span>
+              <a class="button-login" href="#">Выйти</a>
+            </div>
+          </div>
+          <ul class="navigation-user">
+            <li class="navigation-user-item">
+              <button class="navigation-user-button">Сравнить</button>
+            </li>
+            <li class="navigation-user-item">
+              <button class="navigation-user-button">Корзина</button>
+            </li>
+          </ul>
+        </section>
+
+        <section class="site-navigation">
+          <button class="button-drop-menu">
+            <span>Каталог товаров</span>
+            <img src="" alt="Кнопка выпадающего меню">
+          </button>
+          <ul class="navigation-list">
+            <li class="navigation-item">
+              <a href="#" class="navigation-link">Доставка</a>
+            </li>
+            <li class="navigation-item">
+              <a href="#" class="navigation-link">Гарантия</a>
+            </li>
+            <li class="navigation-item">
+              <a href="#" class="navigation-link">Контакты</a>
+            </li>
+          </ul>
+        </section>
+
+      </nav>
+    </header>
```
Хедер после правок перенеси с индекса

yury-dergachev commented on catalog.html:
```diff
@@ -0,0 +1,257 @@
+
+    <main class="main-inner">
+      <header class="inner-header">
+        <h1 class="inner-header-title">Каталог</h1>
```
`Моноподы для селфи` - не хватает текстовки

yury-dergachev commented on catalog.html:
```diff
@@ -0,0 +1,257 @@
+            <legend class="catalog-filter-title">Цвет</legend>
+            <ul class="catalog-filter-list">
+              <li class="catalog-filter-item">
+                <label class="control">
+                  <input class="control-input" type="checkbox" name="black" checked>
+                  Черный
+                </label>
```
```
<label class="control">
    <input class="control-input" type="checkbox" name="black" checked>
    Черный
  </label>
```
Где-то в индексе был вариант записи пары инпут + лейбл рядом. Давай его придерживаться. Меньше вложеностей - легче читаймость

```
<div>
    <label for="" class="control">Черный</label>
    <input id ="" class="control-input" type="checkbox" name="black" checked>
</div>
```

yury-dergachev commented on catalog.html:
```diff
@@ -0,0 +1,257 @@
+          <button class="button" type="submit">Показать</button>
+        </form>
+
+        <select class="select-control" aria-label="Сортировка">
+          <option value="popular">Популярные</option>
+          <option value="cheap">Дешёвые</option>
+          <option value="expensive">Дорогие</option>
+        </select>
+        <a href="#" class="button-navigation">
+          <span class="visually-hidden">Показать cнизу вверх</span>
+        </a>
+        <a class="button-navigation" href="#">
+          <span class="visually-hidden">Показать сверху вниз</span>
+        </a>
```
селект и ссылки надо дополнительно обернуть в див. Также еще обрати внимание, тут две колонки. Нужна будет общая обертка

![image](https://user-images.githubusercontent.com/77338210/217367650-a89775d4-6480-4145-9ec6-6b1806799d4f.png)

yury-dergachev commented on catalog.html:
```diff
@@ -0,0 +1,257 @@
+              <a href="#">Доставка</a>
+            </li>
+            <li class="footer-navigation-item">
+              <a href="#">Гарантия</a>
+            </li>
+            <li class="footer-navigation-item">
+              <a href="#">Контакты</a>
+            </li>
+          </ul>
+        </nav>
+        <ul class="footer-social">
+          <li class="footer-social-item">
+            <a href="https://t.me/htmlacademy" class="button-social">
+              <span class="visually-hidden">Телеграм</span>
+            </a>
+          </li>
+          <li class="footer-social-item">
+            <a href="https://www.youtube.com/user/htmlacademyru" class="button-social">
+              <span class="visually-hidden">Youtube</span>
+            </a>
+          </li>
+          <li class="footer-social-item">
+            <a href="https://twitter.com/htmlacademy_ru" class="button-social">
+              <span class="visually-hidden">Твиттер</span>
+            </a>
+          </li>
+        </ul>
+      </section>
+
+      <section class="footer-contacts">
+        <h2 class="visually-hidden">Контакты</h2>
+        <address class="footer-contacts-phone">
+          <span>Тел.:</span>
+          <a class="contacts-phone" href="tel:+78128121212">+7 (812) 812-12-12</a>
+        </address>
+        <a href="https://htmlacademy.ru/">
+          <img class="logo-html.academy" src="" alt="Логотип платформы html.academy">
+        </a>
+      </section>
+    </footer>
```
Футер копернуть потом с индекса не забыть после правок

yury-dergachev commented:
Проставь еще для src фейковые адреса, Можно #, чтобы не ругался валидатор. 

Также еще необходимо добавить `container`, который будет выступать в качестве центровщика для контентной части сайта.

По вопросам пиши в телегу, помогу с с разметкой. Тебе написать не могу.

YaTyulen committed:
Убирает обёртку секцией логотипа в хэдере

YaTyulen committed:
Убирает div с классом search-and-login в хэдере

YaTyulen committed:
Исправляет section user-tools

Убирает div и img для иконки и оборачивает ссылку "Войти" в список. Исправляет названия некоторых классов

YaTyulen committed:
Добавление тега label перед инпутом в хэдере

YaTyulen committed:
Исправление названий классов в хэдере

YaTyulen committed:
Убирает img в кнопке выпадающего меню

YaTyulen committed:
Исправление названия класса в main

YaTyulen committed:
Исправление названия классов

YaTyulen committed:
Исправление названий классов

YaTyulen committed:
Убирает img иконку

YaTyulen committed:
Убирает иконку

YaTyulen committed:
Убирает img иконку

YaTyulen committed:
Оборачивает блоки "О нас" и "Контакты" в общую секцию

YaTyulen committed:
Изменение section на div в футере

YaTyulen committed:
Оборачивание картинки в ссылку

YaTyulen committed:
Добавление в комментарий кода для смены разметки слайдеров

YaTyulen committed:
Изменения хэдера на странице каталога после правок в index.html

YaTyulen committed:
Исправляет лейблы перед импутами

YaTyulen committed:
Оборачивает select и ссылки в div

YaTyulen committed:
Оборачивание фильтра и списка продуктов в отдельные секции

YaTyulen committed:
Исправление футера и добавление его в catalog.html

YaTyulen committed:
Изменение заголовка h1 в каталоге

YaTyulen committed:
Внесение правок в слайдер и добавление контентного контейнера

YaTyulen committed:
Изменение слайдера и правка lablel в хэдере

YaTyulen committed:
Исправляет опечатку в имени класса

YaTyulen committed:
Исправляет имя класса

YaTyulen committed:
Исправление месторасположения контейнера в майне в index.html

YaTyulen committed:
Исправление месторасположения контейнера в main в catalog.html
